engine_mgo.go

```diff
@@ -112,6 +112,9 @@ func (eng *mongoEngine) Load(collection Collection, id string) (bool, Document,
 		return false, Document{}, err
 	}
 	cleanup(content)
+	if eng.isBinaryId {
+		content["uuid"] = id
```

Comment:
ATM `isBinaryId` is equivalent to using the `native-store`, and things that are read from the `native-store` always have a `uuid` field. When we decide to store all our data in a consistent manner, or use only one way to read/write directly from the DB this can be generalized.